app/src/main/java/org/mozilla/fenix/browser/browsingmode/BrowsingModeManager.kt

```diff
@@ -31,14 +29,18 @@ interface BrowsingModeManager {
     var mode: BrowsingMode
 }
 
+/**
+ * Wraps a [BrowsingMode] and executes a callback whenever [mode] is updated.
+ */
 class DefaultBrowsingModeManager(
-    private val settings: Settings,
+    private var _mode: BrowsingMode,
```

Comment:
nit: I'd prefer `browsingMode` here instead of `_mode`.

Reply:
Prefixing with an underscore is a [Kotlin convention for backing properties](https://kotlinlang.org/docs/reference/coding-conventions.html#names-for-backing-properties), which I think is appropriate here.  Lmk if you still disagree and we can talk it over.